### ISSUE TITLE
2019-7-29

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#70]
211.Compute grad and numgrad; let us separate the target function; done; what is compute numerical gradient? Compute cost; done; write compute numerical gradient; change octave p to i; done; combine two columns; give up; what is norm in octave? Use it in numpy; done; (2019-6-22)
212.The second two column is different; why octave result is different from python? Compare theta1 and theta2; theta1 is different; problem is in debugInitializeWeights; it seems that octave allocate according to the column; let us use another way to allocate i; done; (2019-6-22)
213.Check theta1; theta1 is correct, now; check theta2; theta2 is correct; check X; X is correct; check y; done; check nn_params, wrong; the key is flatten(‘F’); done; check grad; grad become correct; does grad has something to do with lambda? Yes, rewrite compute gradient; done; (2019-6-22)
214.check computeNumericalGradient; write computeCost; done; copy back to original; done; deal with cost; add flatten(‘F’), done; 
215.It = 1; 0.32068323946855815；It =6; 0.3196604712926606
216.Let us try max iteration on linear regression; write predict; what is pred? 
217.0, 0；1, 10；2, 10；10, 9.88; finally; (2019-6-22)
218.Can we read mat file? Yes; try to load Theta1 and Theta2 from mat file; done; read X and Y from mat file; done; delete extra file; done; (2019-6-23)
219.Can we use our neural network to deal with mnist? Mnist is also handwriting; let us try to use train set in tensorflow; OK; we can see that every time the result is different; done; what is test set and validation set? Validation set deal with hyperparameters; done; (2019-6-23)
